File: DSA_practice/Others/InsertionSort.java
public class InsertionSort {
    public static void main(String[] args) {
        int a[] = new int[]{1, 5, 8, 9, 2, 7, 3};
        System.out.println("Before sorting");
        for (int i = 0; i < a.length; i++) {
            System.out.print(a[i]+" ");
        }
        System.out.println("\nAfter sorting");
        Sort(a, 7);
        for (int i = 0; i < a.length; i++) {
            System.out.print(a[i]+" ");
        }
    }
    public static void Sort(int arr[], int n){
        for(int i=0; i<=n-1; i++){
            int j = i;
            while(j>0 && arr[j]< arr[j-1]){
                int temp = arr[j];
                arr[j] = arr[j-1];
                arr[j-1] = temp;
                j--;
            }
        }
    }
}

File: Algorithms_Analysis/insertionSort.c
#include<stdio.h>
int main(){
    int arr[] = {5, 4, 3, 2, 1, 55};
    // Print array before sorting
    printf("Before Sorting\n");
    for(int i=0; i<6; i++){
        printf("%d ", arr[i]);
    }
    printf("\n");

    for(int i=1; i<6; i++){
        int temp = arr[i];
        int ptr = i-1;
        while(temp < arr[ptr] && ptr>= 0){
            arr[ptr+1] = arr[ptr];
            ptr -= 1;
        }
        arr[ptr+1] = temp;
    }

    // Print array after sorting
    printf("After Sorting\n");
    for(int i=0; i<6; i++){
        printf("%d ", arr[i]);
    }
    return 0;
}